src/components/combo/operations/group.ts

```diff
@@ -46,11 +46,13 @@ export default class GroupDataOperation<T extends object> {
       groupBy(data, (item) => item.value[groupKey] ?? 'Other')
     );
 
-    return groups.flatMap(([group, items]) => {
-      items.sort((a, b) =>
-        this.compareObjects(a.value, b.value, displayKey!, direction)
-      );
+    if (direction !== 'none') {
+      groups.sort((a, b) => {
+        return this.orderBy.get(direction)! * a[0].localeCompare(b[0]);
+      });
+    }
```

Comment:
```suggestion
    if (direction !== 'none') {
      const orderBy = this.orderBy.get(direction)!;
      groups.sort((a, b) => orderBy * comparer.compare(a[0], b[0]));
    }
```
Similar to the other PR you can use an `Intl.Collator` object here for the comparison.

Also, since items are no longer sorted, you can pretty much remove all the code above, which manages resolving item keys and comparisons.